tests/unit/expire.tcl

```diff
@@ -601,4 +601,85 @@ start_server {tags {"expire"}} {
            {del foo}
         }
     } {} {needs:repl}
+
+    test {EXPIRE with NX option on a key with ttl} {
+        r SET foo bar EX 100
+        r EXPIRE foo 200 NX
+    } {0}
+
+    test {EXPIRE with NX option on a key without ttl} {
+        r SET foo bar
+        r EXPIRE foo 200 NX
+    } {1}
+
+    test {EXPIREAT with NX option on a key with ttl} {
+        r SET foo bar EX 100
+        r EXPIREAT foo 200 NX
+    } {0}
+
+    test {EXPIREAT with NX option on a key without ttl} {
+        r SET foo bar
+        r EXPIREAT foo 200 NX
+    } {1}
+
+    test {PEXPIRE with NX option on a key with ttl} {
+        r SET foo bar EX 100
+        r PEXPIRE foo 200 NX
```

Comment:
to avoid timing issues, when we work with milliseconds, let's use values such as 200,000 (200 seconds).